### PR TITLE
[1.17.x] Add methods to generate blockstate and models for button, pressure plate, sign

### PR DESCRIPTION
**Pull request by dipo33**
Data and asset generators are a great thing. But I miss the lack of method to generate block state and models for button (and other vanilla blocks too). This PR adds this methods to generate block states and models for button easily. If this PR gets approved, expect few other similar PRs about adding generator methods for vanilla blocks such as pressure plate.

EDIT: At this point the PR now adds methods for generating block states and models also for pressure plate and sign.

### CODE REVIEW

File: src/main/java/net/minecraftforge/client/model/generators/BlockStateProvider.java [hunk 1]
```diff
@@ -423,7 +445,7 @@ public void wallBlock(WallBlock block, String name, ResourceLocation texture) {
     private void wallBlockInternal(WallBlock block, String baseName, ResourceLocation texture) {
         wallBlock(block, models().wallPost(baseName + "_post", texture), models().wallSide(baseName + "_side", texture), models().wallSideTall(baseName + "_side_tall", texture));
     }
-    
+
```
Comment (ShroomAgent27) [inline src/main/java/net/minecraftforge/client/model/generators/BlockStateProvider.java]: Could you remove this whitespace change?

File: src/main/java/net/minecraftforge/client/model/generators/BlockStateProvider.java [hunk 2]
```diff
@@ -442,7 +464,7 @@ public void wallBlock(WallBlock block, ModelFile post, ModelFile side, ModelFile
         		wallSidePart(builder, sideTall, e, WallSide.TALL);
         	});
     }
-    
+
```
Comment (ShroomAgent27) [inline src/main/java/net/minecraftforge/client/model/generators/BlockStateProvider.java]: Could you remove this whitespace change?

Comment (dipo33) [reply src/main/java/net/minecraftforge/client/model/generators/BlockStateProvider.java]: Done

File: src/main/java/net/minecraftforge/client/model/generators/BlockStateProvider.java (hunk 1 above)
Comment (dipo33) [reply src/main/java/net/minecraftforge/client/model/generators/BlockStateProvider.java]: Done

**Comment by ShroomAgent27** [comment]
I wrote a [test mod](https://github.com/FoalFreePress/TestingForgeMods/tree/a5e4b3ade53908c2fbf353ca842016d40edbd69e/8139/src/main/java/com/example/examplemod) that adds 4 modded buttons and they all look good to me.

**Review by ShroomAgent27 — APPROVED** [review]

**Comment by ShroomAgent27** [comment]
>  If this PR gets approved, expect few other similar PRs about adding generator methods for vanilla blocks such as pressure plate.

Could you edit this PR to include these features now, actually? That way you only one PR is needed in order to get all of these features implemented.

**Comment by dipo33** [comment]
Alright, will do

**Comment by dipo33** [comment]
I don't really know what else to add, if you have any suggestions what would deserve an own model / block state generator, tell me. Otherwise consider it done.

**Comment by ShroomAgent27** [comment]
[Looks good to me.](https://github.com/FoalFreePress/TestingForgeMods/tree/121290fa360721f77c01da586834aacca288a4c8/8139/src/main/java/com/example/examplemod)


Edit: Could you edit the PR's title and description to more accurately reflect the content in the PR?

**Review by ShroomAgent27 — APPROVED** [review]

**Comment by dipo33** [comment]
Done

**Pull request merged**